### PR TITLE
fix(TDI-45899): Delete custom transactional record.

### PR DESCRIPTION
**Pull request by dmytrogrygorenko**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-45899

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [X] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/749/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/749/)

File: components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/output/NsObjectOutputTransducer.java [hunk 1]
```diff
@@ -171,6 +172,13 @@ public Object write(IndexedRecord indexedRecord) {
                 nullFieldNames.remove("typeId");
                 writeSimpleField(nsObject, typeIdFieldDesc.asSimple(),
                         false, nullFieldNames, customizationRef.getInternalId());
+            } else if (recordTypeInfo.getRefType() == RefType.CUSTOM_TRANSACTION_REF) {
+                CustomTransactionTypeInfo customTransRecordTypeInfo = (CustomTransactionTypeInfo) recordTypeInfo;
```
Comment (mbasiuk-talend) [inline components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/output/NsObjectOutputTransducer.java:176]: why do you need casting here?
Could you get Name directly from recordTypeInfo?

File: components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/output/NsObjectOutputTransducer.java [hunk 2]
```diff
@@ -171,6 +172,13 @@ public Object write(IndexedRecord indexedRecord) {
                 nullFieldNames.remove("typeId");
                 writeSimpleField(nsObject, typeIdFieldDesc.asSimple(),
                         false, nullFieldNames, customizationRef.getInternalId());
+            } else if (recordTypeInfo.getRefType() == RefType.CUSTOM_TRANSACTION_REF) {
```
Comment (mbasiuk-talend) [inline components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/output/NsObjectOutputTransducer.java:175]: Could you please add test for this issue.

File: components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/output/NsObjectOutputTransducer.java (hunk 1 above)
Comment (dmytrogrygorenko) [reply components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/output/NsObjectOutputTransducer.java]: Fixed

File: components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/output/NsObjectOutputTransducer.java (hunk 2 above)
Comment (dmytrogrygorenko) [reply components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/output/NsObjectOutputTransducer.java:175]: Test was created

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/785/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/785/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/786/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/786/)

**Review by mbasiuk-talend — APPROVED** [review]
LGTM

**Pull request merged**